packages/preferences-persistence/src/migrations/preferences-package-data/convert-editor-settings.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Internal dependencies
+ */
+
+export default function convertEditorSettings( data ) {
+	let newData = data;
+	const settingsToMoveToCore = [ 'allowRightClickOverrides' ];
+
+	settingsToMoveToCore.forEach( ( setting ) => {
+		if ( data?.[ 'core/edit-post' ]?.[ setting ] !== undefined ) {
+			newData = {
+				...newData,
+				core: {
+					...newData?.core,
+					[ setting ]: data[ 'core/edit-post' ][ setting ],
+				},
+			};
+			delete newData[ 'core/edit-post' ][ setting ];
+		}
+	} );
+
+	if ( Object.keys( newData?.[ 'core/edit-post' ] ?? {} )?.length === 0 ) {
+		delete newData[ 'core/edit-post' ];
+	}
```

Comment:
For example, does it make sense to remove the same setting in the `core/edit-site` store as shown below?

```javascript
delete newData[ 'core/edit-site' ][ setting ];
if ( Object.keys( newData?.[ 'core/edit-site' ] ?? {} )?.length === 0 ) {
	delete newData[ 'core/edit-site' ];
}

```

Reply:
Good catch, forgot to do that.